nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/WriteAheadFlowFileRepository.java

```diff
@@ -200,17 +211,19 @@ protected void initialize(final ResourceClaimManager claimManager, final Reposit
         // delete backup. On restore, if no files exist in partition's directory, would have to check backup directory
         this.serdeFactory = serdeFactory;
 
-        if (walImplementation.equals(SEQUENTIAL_ACCESS_WAL)) {
+        // The specified implementation can be plaintext or encrypted; the only difference is the serde factory
+        if (isSequentialAccessWAL(walImplementation)) {
+            // TODO: May need to instantiate ESAWAL for clarity?
```

Comment:
This todo seems not necessary.  Given the todos related to follow on work with NIFI-6617 though this seems fine for now to help get more evaluation cycles on this new capability.